core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -233,12 +252,56 @@ public static boolean defaultUseHttp1Pipelining() {
      * {@code defaultBackoffSpec} in the constructor call.
      *
      * <p>The default value of this flag is {@value DEFAULT_DEFAULT_BACKOFF_SPEC}. Specify the
-     * {@code -Dcom.linecorp.armeria.defaultBackoffSpec=<spec>} to override the default value.
+     * {@code -Dcom.linecorp.armeria.defaultBackoffSpec=<spec>} JVM option to override the default value.
      */
     public static String defaultBackoffSpec() {
         return DEFAULT_BACKOFF_SPEC;
     }
 
+    /**
+     * Returns the value of the {@code routeCache} parameter. It would be used to create a route {@link Cache}
+     * instance using {@link Caffeine#from(String)}. The {@link Cache} would hold the mappings of
+     * {@link PathMappingContext} and the designated {@link ServiceConfig} for a request to improve
+     * server performance.
+     *
+     * <p>The default value of this flag is {@value DEFAULT_ROUTE_CACHE_SPEC}. Specify the
+     * {@code -Dcom.linecorp.armeria.routeCache=<spec>} JVM option to override the default value.
+     * Also, specify {@code -Dcom.linecorp.armeria.routeCache=off} JVM option to disable it.
+     */
+    public static Optional<String> routeCacheSpec() {
+        return ROUTE_CACHE_SPEC;
+    }
+
+    /**
+     * Returns the value of the {@code compositeServiceCache} parameter. It would be used to create a
+     * service {@link Cache} instance using {@link Caffeine#from(String)}. The {@link Cache} would hold
+     * the mappings of {@link PathMappingContext} and the designated {@link CompositeServiceEntry} for
+     * a request to improve server performance.
+     *
+     * <p>The default value of this flag is {@value DEFAULT_COMPOSITE_SERVICE_CACHE_SPEC}. Specify the
+     * {@code -Dcom.linecorp.armeria.compositeServiceCache=<spec>} JVM option to override the default value.
+     * Also, specify {@code -Dcom.linecorp.armeria.compositeServiceCache=off} JVM option to disable it.
+     */
+    public static Optional<String> compositeServiceCacheSpec() {
+        return COMPOSITE_SERVICE_CACHE_SPEC;
+    }
+
+    private static Optional<String> caffeineSpec(String name, String defaultValue) {
+        final String spec = get(name, defaultValue, value -> {
+            try {
+                if (!"off".equals(value)) {
+                    CaffeineSpec.parse(value);
+                }
+                return true;
+            } catch (Exception e) {
+                logger.error("Invalid cache spec: {}", e);
```

Comment:
- ` {}` is unnecessary.
- I guess it's probably OK to skip logging this, because we log the effective property value and why it's used.